Added backport process documentation (#3407)
elastic/eui: wiki/backport-process.md

# EUI backport patches

In general, we strongly encourage updating to the latest version of EUI to obtain bug fixes, and we do not actively consider backporting fixes to previous major or minor release versions. The exception to this is when supporting Kibana's release process, as we want to avoid pushing larger changes near the feature freeze.

### Tracking issue

When preparing for a backport a github issue in should be created in EUI referencing the relevant issues and/or PRs to be included - see https://github.com/elastic/eui/issues/3386 as an example. This issue is used to keep track of the patch's completion progress and to ensure the desired changes are included in the release.

### Patch process

> This provides a walkthrough of the patching & backport release process; examples are taken from the release of x.x.x based on https://github.com/elastic/eui/issues/3386

Reviewer: ```suggestion
This provides a walkthrough of the patching & backport release process; examples are taken from the release of v22.3.1 based on https://github.com/elastic/eui/issues/3386
```

Author: gah, thank you

* Update your local repo's knowledge of commits so they can be referenced - `git checkout master && git pull`

> * Identify the target version of EUI to patch; GH issue says the new version should be `22.3.1` and I confirmed the patch's base is `22.3.0`

Reviewer: ```suggestion
* Identify the target version of EUI to patch; GitHub issue says the new version should be `22.3.1` and I confirmed the patch's base is `22.3.0`
```

  * in the EUI git repo, checkout the release tag the patch is intended for - `git checkout v22.3.0`
  * create a new branch from the versioned tag, the name is unimportant but I use the target version without a leading `v` - `git checkout -b 22.3.1`
* Run `yarn` to ensure you have the correct dependencies for that point in time installed
* Apply the commit(s) with the desired changes

>   * GH issue references #3369, #3378, #3330, and #3398

Reviewer: ```suggestion
  * GitHub issue references #3369, #3378, #3330, and #3398
```

* Start the dev server and check that the intended changes have been properly applied, you don't want to repeat this process to patch the patch - `yarn start`
* Once everything looks correct, it's time to release; the `yarn release` script only works when releasing from `master`, so we'll run [a subset of those steps](https://github.com/elastic/eui/blob/06fc9a6880766168aec1a622873e7f6fe1b3d42b/scripts/release.js#L34-L57) manually
    * Run the unit tests again - `npm test`
    * Create the release builds - `npm run build`
    * Update the I18n tokens - `npm run update-token-changelog -- patch`
    * Use npm to update package.json & package-lock.json version, git commit, and git tag - `npm version patch`
    * Push the version commit & tag to upstream - `git push upstream --tags`
    * Publish the new version to npm
        * Get your npm One Time Password (OTP) from Google Authenticator, Authy, etc
        * `npm publish --otp=your-one-time-password`
* Update `master`'s changelog to include this release
    * On the branch you used to build & release, copy the relevant changelog section - e.g. contents of `## [`22.3.1`](https://github.com/elastic/eui/tree/v22.3.1)`

>     * Checkoug `master` - `git checkout master`

Reviewer: ```suggestion
    * Checkout `master` - `git checkout master`
```
